internal/provider/atlas_schema_data_source_test.go

```diff
@@ -1,16 +1,32 @@
 package provider_test
 
 import (
+	"context"
+	"reflect"
 	"testing"
 
+	"ariga.io/ariga/terraform-provider-atlas/internal/provider"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/zclconf/go-cty/cty"
 )
 
 const testAccData = `
 data "atlas_schema" "market" {
-  dev_db_url = "mysql://root:pass@localhost:3307/test"
-  src = <<-EOT
+	dev_db_url = "mysql://root:pass@localhost:3307/test"
+	variables = [
+		yamlencode({
+			tenant = "bar",
```

Comment:
can you add tests to verify other supported types are handled well? 
bool, int, list(int)..